### ISSUE TITLE
EditorBuilder registers deprecated PhpSearchStrategy

### ISSUE THREAD

**Issue by davidgorges** [gnugat/redaktilo#120]
The triggered ``\E_USER_DEPRECATED`` errors cause phpunit to fail.

If anyone else runs into that:
 ``\PHPUnit_Framework_Error_Deprecated::$enabled = FALSE;``
disables the test fails.

**Comment by gnugat** [comment]
Hi @davidgorges thanks for your report. I'll remove `PhpSearchStrategy` from `EditorBuilder` to avoid this notice.

**Comment by WouterJ** [comment]
@gnugat it can't be removed, as it's there for BC reasons. I think there are 2 options:
1. Instead of adding it as first search strategy, add it as last, so with normal usage, the notice won't be triggered (initializing the class doesn't trigger the deprecation notice)
1. Remove the trigger from `PhpSearchStrategy#supports()`. Again, this would avoid getting the error, unless someone uses the PhpSearchStrategy explicitely. In that case, it should get this notice.

**Comment by gnugat** [comment]
@WouterJ good point, I'll try your first advice, if it doesn't do the trick I'll check the second :)

**Comment by gnugat** [comment]
@davidgorges could you provide us with a small piece of code to reproduce the issue? I'd like to add a test for it.

**Comment by WouterJ** [comment]
@gnugat I think this will do (didn't test it though):

```php
$editor = EditorFactory::createEditor();
$editor->findBelow('something');
```

**Comment by gnugat** [comment]
That's weird, `example` is full of tests that use `EditorFactory::createEditor()` + `$editor#jumpBelow()` and we never had a failure caused by PHP notice...

**Comment by WouterJ** [comment]
@gnugat try with a line number search pattern. As that's the only strategy registered *after* the PHP search strategy (I forget about the priorty argument my previous comments). https://github.com/gnugat/redaktilo/blob/master/src/Gnugat/Redaktilo/Service/EditorBuilder.php#L86-L89

With all other patterns, there is a matching strategy stopping the iteration.

**Comment by gnugat** [comment]
@WouterJ good point! LineNumberSearchStrategy is also deprecated, so I've added a test checking the notice (should be about LineNumberSearchStrategy and not PhpSearchStrategy).
